wts/resources/oauth2.py

```diff
@@ -71,12 +77,26 @@ def refresh_refresh_token(tokens):
             bytes(refresh_token, encoding="utf8")
         )
 
+    # get the username of the current logged in user.
+    # `current_user` validates the token and relies on `OIDC_ISSUER`
+    # to know the issuer
+    client, _ = get_oauth_client(idp="default")
+    flask.current_app.config["OIDC_ISSUER"] = client.api_base_url.strip("/")
+    user = current_user
```

Comment:
Can you explain how this code works? My understanding is that `current_user` is an alias for the `set_current_user` function in authutils, but how does it validate the token and read an env variable without the function being called? 👀

Reply:
yeah it actually doesn't fail during `user = current_user` but during `username = user.username`

`current_user` find the user by checking the information in the token. If the token was generated by another issuer than the one authutils expects, the token is rejected.

The `CurrentUser` initialization calls `validate_request()`, which calls `validate_jwt()`, and the relevant line is here: https://github.com/uc-cdis/authutils/blob/4.0.0/authutils/token/validate.py#L207